crd-generator/api-v2/src/test/java/io/fabric8/crdv2/generator/v1/JsonSchemaDefaultValueTest.java

```diff
@@ -55,24 +58,28 @@ private static final class ClassInTest {
     @JsonProperty(defaultValue = "1337")
     int defaultValueForInt;
 
-    @JsonProperty(defaultValue = "1337L")
+    @JsonProperty(defaultValue = "1337")
     long defaultValueForLong;
 
     @JsonProperty(defaultValue = "13.37")
     float defaultValueForFloat;
 
-    @JsonProperty(defaultValue = "13.37d")
+    @JsonProperty(defaultValue = "13.37")
     double defaultValueForDouble;
+
+    @JsonProperty
+    @Default("[]")
+    double[] defaultValueForDoubleArray;
   }
 
   @Test
-  @DisplayName("JsonProperty default value should take precedence over Default annotation")
+  @DisplayName("Default annotation should take precedence over JsonProperty")
```

Comment:
https://github.com/fabric8io/kubernetes-client/pull/6666#discussion_r1861130328

We're changing the order of precedence here.
I'm fine with this, but just want to make sure why we're doing this and to have some paper trail in case we want to trace this back in the future.

Reply:
> We're changing the order of precedence here.

Yes back to what it was in the initial port, and inline with the other annotations - the fabric8 ones are dominant.

Looking back over the last fix the switch in ordering may have been done because I was initially wondering if we should deprecate at least some of our fabric8 annotations as you can do some of the same things now with the jackson annotations - but in many cases there's a potential semantic mismatch where the jackson annotation value may not really be appropriate for a crd.  That brings up another thought - if we don't trust the JsonProperty default to be yaml or json and don't want a hard error, then this fix will need refined a little so that there will be a warning instead.